### PR TITLE
Modified load_Intermediate_PregnantAndBreastfeeding script

### PR DESCRIPTION
**Pull request by DennisGibz**

### CODE REVIEW

File: Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql [hunk 1]
```diff
@@ -260,56 +261,74 @@ select				SiteCode
...
 	  ,PatientPKHash
       ,[IsPregnant]
       ,[IsBreastFeedingAsOfDate]
       ,[AsOfDate]
-      ,[VisitDate]
+      ,BreastFeedingRelatedVisitDate
+	  ,PregnancyRelatedVisitDate
       ,[IsPBFW]
-  FROM PBFW
+  FROM PBFW where [SiteCode] is not null
```
Comment (nobert-mumo) [inline Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:328]: @DennisGibz why would this happen? Could we be omitting genuine clients ?

File: Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql [hunk 2]
```diff
@@ -260,56 +261,74 @@ select				SiteCode
...
-	   SELECT SiteCode
-			,PatientPK
-			,PatientPKHash
-			,IsPregnant =1
-			,IsBreastFeedingAsOfDate =0
-			,AsOfDate
-			,VisitDate
-			,IsPBFW	 =1   
-	   FROM PregnantAsOfDate
+	   from MaxOrderedCombineBreastFeeding
```
Comment (nobert-mumo) [inline Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:307]: Nice one with the full join. Just curious how did we handle the duplicates we were getting even after changing from `union` to `full join` ?

File: Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql (hunk 1 above)
Comment (nobert-mumo) [reply Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:328]: @DennisGibz, this is still pending right ?

**Review by nthusi-codes — CHANGES_REQUESTED** [review]

File: Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql (hunk 1 above)
Comment (DennisGibz) [reply Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:328]: @nobert-mumo I don't think they are genuine records. I have re-run the script step by step and seems like the full outer join is causing the nulls.
See sample outputs below;

image

File: Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql (hunk 2 above)
Comment (DennisGibz) [reply Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:307]: The full outer join works well but to remove dups but the unmapped patients are left with nulls. Thats why I am using the where sitecode is not null to eliminate the nulls

File: Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql [hunk 3]
```diff
@@ -227,17 +228,17 @@ IsBreastFeedingFromHeiDOB As( ---breastfeeding confirmed from MNCH
...
 						,0 IsPBFW 
 				From MaxOrderedBreastFeedingAsOfDate                           
 				left join ods.dbo.MNCH_MotherBabyPairs pairs
 				on MaxOrderedBreastFeedingAsOfDate.SiteCode = pairs.SiteCode  and MaxOrderedBreastFeedingAsOfDate.PatientPK = pairs.MotherPatientPK
 				left join ods.dbo.MNCH_Patient  Patient
 				on pairs.SiteCode = Patient.SiteCode 
 							and pairs.BabyPatientPK = Patient.PatientPK
-				WHERE Datediff(month,Patient.DOB,@as_of_date) <=24 
+				WHERE Datediff(month,Patient.DOB,@as_of_date) <=24 ---and pairs.PatientPK = 12706 and pairs.SiteCode = 11259	
 ),
```
[the remark this answers is not in the document]
Comment (DennisGibz) [reply Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:240]: Well noted @nthusi-codes . I'll clean up the script.

File: Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql (hunk 2 above)
Comment (nobert-mumo) [reply Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:308]: @DennisGibz not sure I follow, coz with full join, I expected we would have coalesce in the column section to take care of nulls. 

e.g. `coalesce(MaxOrderedCombineBreastFeeding.SiteCode, PregnantAsOfDate.SiteCode) as SiteCode`

Also not sure the full join would eliminate the dups ?

File: Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql (hunk 1 above)
Comment (nobert-mumo) [reply Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:329]: @DennisGibz I still think is related to the full join up there.

**Review by nobert-mumo — CHANGES_REQUESTED** [review]

**Review by nobert-mumo — APPROVED** [review]
@DennisGibz apart from the small comment on the duplicate code, majorly LGTM

File: Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql [hunk 4]
```diff
@@ -260,53 +266,77 @@ select				SiteCode
...
-/*
-Step 10 : Insert into [ODS].[dbo].[Intermediate_PregnantAndBreastFeeding] for reuse
-*/
-insert into [ODS].[dbo].[Intermediate_PregnantAndBreastFeeding]([SiteCode],[PatientPK],PatientPKHash,[IsPregnant],[IsBreastFeeding],[AsOfDate],[VisitDate],[IsPBFW])
+--End
+--/*
+--Step 10 : Insert into [ODS].[dbo].[Intermediate_PregnantAndBreastFeeding] for reuse
+--*/
+insert into [ODS].[Intermediate].[Intermediate_PregnantAndBreastFeeding]([SiteCode],[PatientPK],PatientPKHash,[IsPregnant],[IsBreastFeeding],[AsOfDate],BreastFeedingRelatedVisitDate,PregnancyRelatedVisitDate,[IsPBFW])
+insert into [ODS].[Intermediate].[Intermediate_PregnantAndBreastFeeding]([SiteCode],[PatientPK],PatientPKHash,[IsPregnant],[IsBreastFeeding],[AsOfDate],BreastFeedingRelatedVisitDate,PregnancyRelatedVisitDate,[IsPBFW])
```
[the remark this answers is not in the document]
Comment (DennisGibz) [reply Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:328]: Not sure how you are getting above. Below is a screenshot of the results that I have

image

[the remark this answers is not in the document]
Comment (nobert-mumo) [reply Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:328]: > Not sure how you are getting above. Below is a screenshot of the results that I have
> 
> image

@DennisGibz not really, I meant duplicated code on line 327 and 328, that would throw an error when the table is building right ?

[the remark this answers is not in the document]
Comment (DennisGibz) [reply Scripts/ODS/load_intermediate_tables/CT_intermediate_tables/Load_Intermediate_PregnantAndBreastFeeding.sql:328]: Surprisingly I wasn't throwing any errors...I have removed it though. Thanks

**Review by nthusi-codes — APPROVED** [review]

**Pull request merged**